client/src/pages/course/score.tsx

```diff
@@ -130,6 +137,31 @@ export function Page(props: CoursePageProps) {
     loadInitialData();
   }, []);
 
+  const handleSettings = () => {
+    setIsVisibleSettings(!isVisibleSetting);
+  };
+
+  const handleModalCancel = () => {
+    setIsVisibleSettings(!isVisibleSetting);
+  };
+
+  const handleModalOk = (values: Store) => {
+    setIsNotVisibleColumns(JSON.stringify(Object.keys(values).filter((value: string) => values[value] === false)));
+    setIsVisibleSettings(!isVisibleSetting);
+  };
+
+  const getVisibleTasks = (tasks: CourseTask[]) => {
+    return tasks.map(task => {
+      notVisibleColumns?.includes(task.name) ? (task.isVisible = false) : (task.isVisible = true);
+      return task;
+    });
+  };
```

Comment:
Please try to avoid such constructions like:
```
return {
    return
}
```
When you're using arrow functions you can write shorter (without return), like this:
```
 const getVisibleTasks = (tasks: CourseTask[]) => (
   tasks.map(task => {
      notVisibleColumns?.includes(task.name) ? (task.isVisible = false) : (task.isVisible = true);
      return task;
    })
  );
```

Reply:
also
```
task.isVisible  = notVisibleColumns?.includes(task.name) ? false : true;
```

Reply:
And also it's not good that you modify "task" (except if you know what you do)) )
So robably better:
```
{
   ...task,
   isVisible: !notVisibleColumns?.includes(task.name),
}
```

Reply:
done